InteractivePreGeneratedViews/FileViewCacheFactory.cs

```diff
@@ -79,7 +79,27 @@ protected override void Save(XDocument viewsXml)
 
             lock (_lockObject)
             {
-                viewsXml.Save(_viewFilePath);
+                if (File.Exists(_viewFilePath))
+                {
+                    // Prevent a race condition whereby two threads could call Load(), both receive a null return, then both call Save(). While
+                    // the lock prevents simultaneous saving, the views would still be saved twice in succession, which is unnecessary.
+                    return;
+                }
+
+                try
+                {
+                    viewsXml.Save(_viewFilePath);
+                }
+                catch(IOException e)
+                {
+                    if (e.HResult == unchecked ( (int)0x80070020 ))

```

Comment:
I think, the only thing you can do here is to retry in case of failure and throw after a number of attempts. Again, a problem here may indicate a bigger issue.